Celeste.Mod.mm/Mod/Everest/TextInput.cs

```diff
@@ -18,28 +17,58 @@ public static class TextInput {
                 return;
             Initialized = true;
 
+            // Subscribing is useless as long as we don't call `StartTextInput` so its ok for this to be here
             TextInputEXT.TextInput += ReceiveTextInput;
-            TextInputEXT.StartTextInput();
+            CheckTextStatus(); // Required check to handle pre-init subscriptions
         }
 
         internal static void Shutdown() {
             if (!Initialized)
                 return;
 
-            TextInputEXT.StopTextInput();
+            if (TextInputEXT.IsTextInputActive())
+                TextInputEXT.StopTextInput();
             TextInputEXT.TextInput -= ReceiveTextInput;
         }
 
         internal static void ReceiveTextInput(char c) {
             // Invoke our own event handler.
-            OnInput?.Invoke(c);
+            _OnInput?.Invoke(c);
         }
 
+        private static event Action<char> _OnInput;
+
         /// <summary>
         /// Invoked whenever text input occurs, including some "input action" characters.
         /// Take a look at the FNA TextInputExt documentation for more info: https://github.com/FNA-XNA/FNA/wiki/5:-FNA-Extensions#textinputext
+        /// !!!Make sure to unsubscribe to this as soon as you're done with input, otherwise it'll cause issues with
```

Comment:
unsubscribe "from", not "to". also no exclamation mark spam in documentation please, and this sentence is generally written a bit too casually